File: Inheritance/2.rb
class Cal
  def initialize(v1, v2)
    @v1 = v1
    @v2 = v2
  end
  def add()
    return @v1 + @v2
  end
  def subtract()
    return @v1 - @v2
  end
  def getVal1()
    return @v1
  end
  def getVal2(v2) #로컬 변수 넣으려면, 사용할때도 받아야 함
    return @v2
  end
  def setVal1(v1)
    if v1.is_a?(Integer)
      @v1 = v1
    end
  end
  def setVal2(v2)
    @v2 = v2
  end
end

class CalMultiply < Cal
  def multiply()
    return @v1 * @v2
  end
end

c1 = CalMultiply.new(10,20)
p c1, c1.multiply
p c1, c1.add

File: Inheritance/2.py
#OOP_Variable 3.py에서 복제

class Cal:
    def __init__(self,v1,v2):
        if isinstance(v1, int):
            self.val1 = v1
        if isinstance(v2, int):
            self.val2 = v2
    def add(v1,v2):
        return v1 + v2
    def add2(self):
        return self.val1 + self.val2
    def subtract(v1, v2):
        return v1 - v2
    def subtract2(self):
        return self.val1 - self.val2
    def setVal1(self,v):
        if isinstance(v, int):
            self.val1 = v
    def getVal1(self):
        return self.val1
    def getVal2(self):
        return self.val2


class CalMultiply(Cal):
    def multiply(self):
        return self.val1 * self.val2


class CalDivide(CalMultiply):
    def devide(self):
        return self.val1 / self.val2


c1 = CalMultiply(10,20)
print(c1.multiply())


c2 = CalDivide(100, 20)
print(c2, c2.devide())
